README.md

```diff
@@ -4,19 +4,105 @@ It is an Android application for [Tango](https://get.google.com/tango/)-enabled
 Do not hesitate to give us feedback if something is broken or if you think it lacks some features. The best way to do this is by adding issues to this repository.
 
 # Kickstart
 The app is available in Google's Play Store: https://play.google.com/store/apps/details?id=eu.intermodalics.tango_ros_streamer
 
-# Manual installation
+# Step-by-step instructions
+For this guide, we assume that you have a clean installation of Ubuntu 14.04 and we will use ROS Indigo. However, the app should also work on Ubuntu 16.04 in combination with ROS Kinetic.
+
+## Install ROS
+Follow this [guide](http://wiki.ros.org/indigo/Installation/Ubuntu) and go for the Desktop -- Full Install:
+
+```
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+sudo apt-get update
+sudo apt-get install ros-indigo-desktop-full
+sudo rosdep init
+rosdep update
+echo "source /opt/ros/indigo/setup.bash" >> ~/.bashrc
+source ~/.bashrc
+```
+
+## Building roscpp_android
+Next, we need to build ```roscpp_android```. A more detailed guide is available [here](http://wiki.ros.org/android_ndk/Tutorials/BuildingNativeROSPackages).
+
+```
+sudo apt-get update
+sudo apt-get install curl git
+curl -sSL https://get.docker.com/ | sudo sh
+mkdir ~/ros-android-ndk
+cd ~/ros-android-ndk
+git clone https://github.com/ekumenlabs/roscpp_android.git
+cd roscpp_android
+./do_docker.sh --portable
+```
+
+Note that the last command take will take long time to complete. Once this has ended, copy the the content of the output ```roscpp_android_ndk``` folder into ```tango_ros/third_party/roscpp_android_ndk/```, except the ```Android.mk``` file and the ```share``` folder.
+
+```
+cp -r ~/ros-android-ndk/roscpp_android/output/roscpp_android_ndk/lib/ ~/tango_ros_ws/src/tango_ros/third_party/roscpp_android_ndk/
+cp -r ~/ros-android-ndk/roscpp_android/output/roscpp_android_ndk/include/ ~/tango_ros_ws/src/tango_ros/third_party/roscpp_android_ndk/
+cp ~/ros-android-ndk/roscpp_android/output/roscpp_android_ndk/Application.mk ~/tango_ros_ws/src/tango_ros/third_party/roscpp_android_ndk/
 ```
-sudo apt-get install python-catkin-tools
+
+## Build the dependencies
+```
+sudo apt-get install git python-catkin-tools
 mkdir -p ~/tango_ros_ws/src
 cd ~/tango_ros_ws/src
 git clone --recursive git@github.com:Intermodalics/tango_ros.git
 cd ~/tango_ros_ws
 catkin build
 ```  
-To build and run the app: see [here](https://github.com/Intermodalics/tango_ros/blob/master/TangoRosStreamer/README.md).  
+
+## Android Studio
+The steps detailed below are based on this [installation guide](http://wiki.ros.org/android/kinetic/Android%20Studio/Download).
+
+For Android Studio we need Java, so let's install this first:
+```
+sudo apt-get update
+sudo apt-get install openjdk-7-jdk
```

Comment:
It's not available out of the box on Ubuntu 14.04 which is assumed by this tutorial. Let me add a line saying that you should install openjdk-8-jdk on Ubuntu 16.04 to avoid confusion.